src/modules/SambaNetJoin.pm

```diff
@@ -302,34 +309,33 @@ sub Join {
 	$self->PrepareCTDB ($server);
 
 	my $krb_file	= $tmpdir."/krb5.conf";
-	my $realm	= SambaAD->Realm ();
-	my $content     = "[global]$include\n\trealm = $realm\n\tsecurity = ADS\n\tworkgroup = $domain\n";
+	$glb_overrides{"realm"} = $realm;
+	$glb_overrides{"workgroup"} = $domain;
+	$glb_overrides{"security"} = "ads";
 	my $kerberos_method	= SambaConfig->GlobalGetStr ("kerberos method", "");
 	if ($kerberos_method) {
-	    $content	= $content."\tkerberos method = $kerberos_method\n";
+            $glb_overrides{"kerberos method"} = $kerberos_method;
 	}
         if ($self->ClusterPresent (0)) {
             # ensure cluster related options are used from original file
             # bnc#809208
             my $clustering      = SambaConfig->GlobalGetStr ("clustering", undef);
-            if (defined $clustering) {
-              my $ctdbd_socket    = SambaConfig->GlobalGetStr ("ctdbd socket", "");
-              $content .= "\t" . "clustering = $clustering" . "\n";
-              $content .= "\t" . "ctdbd socket =$ctdbd_socket" . "\n";
-            }
-            else {
+            if (!defined $clustering) {
               y2error ("'clustering' not defined in smb.conf, canceling join attempt");
               return __("Unable to proceed with join: Inconsistent cluster state");
             }
         }
-	SCR->Write (".target.string", $conf_file, $content);
 	$cmd		= "KRB5_CONFIG=$krb_file ";
 	SCR->Write (".target.string", $krb_file, "[realms]\n\t$realm = {\n\tkdc = $server\n\t}\n");
     }
     else {
-	SCR->Write (".target.string", $conf_file, "[global]$include\n\tsecurity = domain\n\tworkgroup = $domain\n");
+	$glb_overrides{"realm"} = undef;
+	$glb_overrides{"workgroup"} = $domain;
+	$glb_overrides{"security"} = "domain";
```

Comment:
How did the previous implementation handle leaving an AD domain, if at all. Is this behaviour affected by this change?